src/main/java/com/sonymobile/tools/gerrit/gerritevents/GerritConnection.java

```diff
@@ -94,6 +94,7 @@
     private AuthenticationUpdater authenticationUpdater = null;
     private final Set<ConnectionListener> listeners = new CopyOnWriteArraySet<ConnectionListener>();
     private int sshRxBufferSize = SSH_RX_BUFFER_SIZE;
+    private StringBuilder eventBuffer = null;
```

Comment:
>  Unfortunately it is not guaranteed that getLine method call is done only by the same thread during keeping this StringBuilder instance

If yes, it would be better to guarantee it somehow or synchronize the method. You could try using the thread-safe `StringBuffer`, but in the current use-case it may become a performance killer. Synchronized `getLine()` would be much better in this case